src/v/kafka/server/connection_context.cc

```diff
@@ -42,6 +43,20 @@ ss::future<> connection_context::process_one_request() {
           if (!sz) {
               return ss::make_ready_future<>();
           }
+          if (sz > _max_request_size()) {
+              vlog(
+                klog.warn,
+                "request from {} is larger ({} bytes) than configured max "
+                "request size {}",
+                _rs.conn->addr,
+                sz,
+                _max_request_size());
+              throw net::invalid_request_error(fmt::format(
```

Comment:
should this return an exceptional future rather than throw since this isn't a coroutine?